packages/ui/src/ui/store/selectors/flow/layout.ts

```diff
@@ -0,0 +1,141 @@
+import {createSelector} from 'reselect';
+import map_ from 'lodash/map';
+import forEach_ from 'lodash/forEach';
+
+import {RootState} from '../../../store/reducers';
+
+export const getFlowLayoutPipelinePath = (state: RootState) => state.flow.layout.pipeline_path;
+export const getFlowLayoutError = (state: RootState) => state.flow.layout.error;
+const getFlowLayoutExpandedComputations = (state: RootState) =>
+    state.flow.layout.expandedComputations;
+const getFlowLayoutExpandedWorkers = (state: RootState) => state.flow.layout.expandedWorkers;
+
+const getFlowLayoutRawData = (state: RootState) => state.flow.layout.data;
+
+type RowData = Pick<
+    ReturnType<typeof getFlowLayoutPartitions>[number],
+    'job' | 'partition' | 'worker'
+>;
+type LayoutExpandable = {
+    $value: Array<Exclude<FlowLayoutDataItem, LayoutExpandable>>;
+    $attributes: {
+        name: string;
+        expanded: boolean;
+        job_count: number;
+        worker_count: number;
+        partition_count: number;
+    };
+};
+
+export type FlowLayoutDataItem = {[K in keyof RowData]: RowData[K] | null} | LayoutExpandable;
+
+export const getFlowLayoutData = createSelector(
+    [getFlowLayoutRawData, getFlowLayoutExpandedComputations, getFlowLayoutExpandedWorkers],
+    (data, expandedComputations, expandedWorkers) => {
+        return (mode: 'computations' | 'workers') => {
+            switch (mode) {
```

Comment:
are you sure you need a switch here? At the moment, there can be only 2 possible values. It would be easier to use if:
`if(mode === 'computations') retrun ...`
`return ....`